apps/studio/src/lib/types/session.ts

```diff
@@ -1,13 +1,22 @@
+import type { Tagged } from "type-fest"
 import { type User } from "@prisma/client"
 import { type IronSession } from "iron-session"
-import { Tagged } from "type-fest"
 
+import type { VerificationToken } from "~/server/modules/database"
 import { type SgidSessionProfile } from "~/server/modules/auth/sgid/sgid.utils"
 
 type CurrentUserId = Tagged<User["id"], "CurrentUserId">
 
 export interface SessionData {
   userId?: CurrentUserId
+  singpass?: {
+    sessionState?: {
+      userId: CurrentUserId
+      verificationToken: VerificationToken
```

Comment:
can i ask why we need the `verificationToken` here cos the singpass sign-in is after?

Reply:
Ah this one is to fit the audit logs, because we would have lost this information by the time they do the Singpass callback, hence opted to store it inside the iron session cookie.